### PR TITLE
Add `must_use` annotations to all futures

### PR DESCRIPTION
**Pull request by ipetkov**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -433,6 +438,7 @@ pub struct ChildStdin {
 /// of a child process. Note that this type is also "futures aware" meaning
 /// that it is both (a) nonblocking and (b) will panic if used off of a
 /// future's task.
+#[must_use = "futures do nothing unless polled"]
```
Comment (alexcrichton) [inline src/lib.rs]: I think this an the other stdio structs aren't futures, right?

**Comment by alexcrichton** [comment]
Good idea!

File: src/lib.rs (hunk 1 above)
Comment (ipetkov) [inline src/lib.rs]: Oops, thought these were adapters akin to `tokio_io`. Will remove from them!

**Comment by ipetkov** [comment]
@alexcrichton Updated as per comments, and rebased!

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**